src/eval.rs

```diff
@@ -252,56 +277,19 @@ fn eval_expr(expr: &Expr, env: &Env, program: &Program) -> Result<Value, Runtime
             let mut env_args = Env::new();
             let function = program.helpers.get(callee).unwrap().clone();
 
-            // check to see proper args
-            if args.len() != function.params.len() {
-                return Err(RuntimeError::IncorrectArgs);
-            }
-            let mut num_matches = 0; // checking to see if we've matched every arg
-
             for i in 0..args.len() {
```

Comment:
do something like `for (arg, param) in args.iter().zip(function.params.iter())` so you don't juggle indices. doing `.get(i).unwrap()` everywhere is a code smell.